### PR TITLE
sql: benchmark pgwire,postgres with SSL

### PR DESCRIPTION
**Pull request by tamird**
postgres ssl config as follows:

```
$ psql -c 'SHOW config_file'
               config_file
-----------------------------------------
 /usr/local/var/postgres/postgresql.conf
(1 row)

$ cat /usr/local/var/postgres/postgresql.conf | grep ssl
ssl = on                # (change requires restart)
ssl_cert_file = '/Users/tamird/src/go/src/github.com/cockroachdb/cockroach/resource/test_certs/node.server.crt'     # (change requires restart)
ssl_key_file = '/Users/tamird/src/go/src/github.com/cockroachdb/cockroach/resource/test_certs/node.server.key'      # (change requires restart)
ssl_ca_file = '/Users/tamird/src/go/src/github.com/cockroachdb/cockroach/resource/test_certs/ca.crt'            # (change requires restart)
```

```
name                   time/op
Bank-4                  864µs ±11%
Select1_HTTP-4          504µs ±19%
Select1_HTTPS-4         463µs ± 7%
Select1_RPC-4           118µs ±12%
Select1_RPCS-4          131µs ± 7%
Select1_Cockroach-4    89.6µs ± 7%
Select1_Postgres-4     80.7µs ± 9%
Select1_MySQL-4        72.8µs ± 9%
Insert1_Cockroach-4    1.45ms ± 4%
Insert1_Postgres-4      199µs ± 7%
Insert10_Cockroach-4   2.03ms ± 6%
Insert10_Postgres-4     248µs ±13%
Insert100_Cockroach-4  6.75ms ± 4%
Insert100_Postgres-4   1.60ms ± 4%
Scan1_Cockroach-4       542µs ± 6%
Scan1_Postgres-4        106µs ± 6%
Scan10_Cockroach-4      619µs ± 8%
Scan10_Postgres-4       110µs ±12%
Scan100_Cockroach-4    1.22ms ± 4%
Scan100_Postgres-4      177µs ± 5%
```

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3336)

### CODE REVIEW

**Comment by petermattis** [comment]
Postgres required certs with bigger keys?

---

Review status: 0 of 14 files reviewed at latest revision, 1 unresolved discussion.

---

*[sql/bench_test.go, line 64 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3336#-K4taXG9MITaTubon67J:-K4taXG9MITaTubon67K:-145570535) ([raw file](https://github.com/cockroachdb/cockroach/blob/97fd1ac6e2188723ece90cc0d6539126cd28366e/sql/bench_test.go#L64)):*
There is a lot of code shared between this and `logic_test.go`. I would guess that we're going to need this boilerplate in other tests as well. Let's move it into a function to create a pgurl from a testserver. Perhaps it deserves to be a method on `TestServer`.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3336#-:-K4tbQwI4ZiDetOS-1W-:-1859784503)*

**Comment by tamird** [comment]
Yeah, it wanted at least 1024 bit certs.

---

Review status: 0 of 15 files reviewed at latest revision, 1 unresolved discussion.

---

*[sql/bench_test.go, line 64 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3336#-K4taXG9MITaTubon67J:-K4tfNBHfQUACF7YmcyE:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/97fd1ac6e2188723ece90cc0d6539126cd28366e/sql/bench_test.go#L64)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3336#-:-K4tfNv5fSa3OVvPLBDH:-1739873451)*

File: sql/bench_test.go [hunk 1]
```diff
@@ -54,7 +52,7 @@ func benchmarkCockroach(b *testing.B, f func(b *testing.B, db *sql.DB)) {
 }
 
 func benchmarkPostgres(b *testing.B, f func(b *testing.B, db *sql.DB)) {
```
Comment (bdarnell) [inline sql/bench_test.go]: Include comments about the necessary postgresql.conf settings in the code, not just the commit message.

**Comment by bdarnell** [comment]
LGTM. Did the switch to larger certs make the regular test suite any slower? Go's RSA key loading is extremely slow (because it does expensive primality tests every time), but I'm not sure if we have enough tests that use TLS that it makes any difference.

**Comment by tamird** [comment]
Review status: 0 of 15 files reviewed at latest revision, 2 unresolved discussions.

---

*[sql/bench_test.go, line 54 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3336#-K4tNdHDihILNyUbYtRo-r2-54:-K4xhfQfYYxQeO6ljqXn:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/9d50560bd60cc6735cd2d63db3581b77d6c0fbf3/sql/bench_test.go#L54)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3336)*

**Comment by tamird** [comment]
> Did the switch to larger certs make the regular test suite any slower?

Doesn't seem to be noticeable.

**Pull request merged**